### DISCUSSION TITLE
Smart Contract wallet integration

### DISCUSSION THREAD

**Discussion by JasoonS** [WalletConnect/walletconnect-monorepo#1980] [Developer Support]
I'm looking to build out a smart contract wallet (with some similarities to Argent or Gnosis Safe), it is very important that we can log into services with wallet connect as that wallet.

It is unclear what the steps are to make this happen reading through the docs. Are there any reference implementations we can take a look at?

**Comment by nazariyv** [comment]
Sorry for the tags, but I am having the same question and am very much interested in the answer:

@pedrouid @ganchoradkov @0xAsimetriq @bkrem
